crates/fs/src/fs.rs

```diff
@@ -561,11 +564,14 @@ impl Fs for RealFs {
                 let mut paths = event
```

Comment:
I don't know which one is more readable. Using `filter` and then `map` or using `filter_map`: 
```rust
let mut paths = event
    .paths
    .iter()
    .filter_map(|path| {
        if path.starts_with(&root_path)
            || (target_root_path.as_ref().is_ok()
                && path.starts_with(target_root_path.as_ref().unwrap()))
        {
            return Some(PathEvent {
                path: path.clone(),
                kind,
            });
        }
        None
    })
    .collect::<Vec<_>>();
```
In terms of performance I would think doing `filter_map` is faster since it does just one loop, but maybe the compiler optimizer `filter` followed by `map`.